msyncd/synchronizer.cpp

```diff
@@ -291,55 +293,27 @@ bool Synchronizer::startScheduledSync(QString aProfileName)
     // All scheduled syncs are online syncs
     // Add this to the waiting online syncs and it will be started when we
     // receive a session connection status from the NetworkManager
-    bool sessionConnected = !iWaitingOnlineSyncs.isEmpty();
-    if(!iWaitingOnlineSyncs.contains(aProfileName))
+    bool accept = acceptScheduledSync(iNetworkManager->isOnline(), iNetworkManager->connectionType());
+    if(accept)
     {
-        iWaitingOnlineSyncs.append(aProfileName);
+        startSync(aProfileName, true);
     }
-    if(!sessionConnected)
+    else if (!iWaitingOnlineSyncs.contains(aProfileName))
     {
-        QObject::connect(iNetworkManager, SIGNAL(connectionSuccess()), this,
-                    SLOT(slotNetworkSessionOpened()), Qt::QueuedConnection);
-        QObject::connect(iNetworkManager, SIGNAL(connectionError()), this,
-                    SLOT(slotNetworkSessionError()), Qt::QueuedConnection);
-        // Request for a network session
-        iNetworkManager->connectSession(true);
+         LOG_DEBUG("Wait for internet connection:" << aProfileName);
+         if (iNetworkManager->isOnline())
+         {
+             LOG_DEBUG("Connection over mobile data plan. The sync will be postponed untill a full connection is available;");
```

Comment:
Renato, are you willing to modify the patch to read these sorts of "configuration settings" from dconf keys or a .ini or something?